include/boost/geometry/strategies/spherical/area_huiller.hpp

```diff
@@ -142,20 +150,14 @@ public :
 
             E = geometry::math::abs(E);
 
-            // In right direction: positive, add area. In left direction: negative, subtract area.
-            // Longitude comparisons are not so obvious. If one is negative, other is positive,
-            // we have to take the dateline into account.
-            // TODO: check this / enhance this, should be more robust. See also the "grow" for ll
-            // TODO: use minmax or "smaller"/"compare" strategy for this
-            calculation_type lon1 = geometry::get_as_radian<0>(p1) < 0
-                ? geometry::get_as_radian<0>(p1) + two_pi
-                : geometry::get_as_radian<0>(p1);
-
-            calculation_type lon2 = geometry::get_as_radian<0>(p2) < 0
-                ? geometry::get_as_radian<0>(p2) + two_pi
-                : geometry::get_as_radian<0>(p2);
+            calculation_type lon = geometry::get_as_radian<0>(p2)
+                                 - geometry::get_as_radian<0>(p1);
+            if ( lon <= 0 )
```

Comment:
Yes I was considering using `lon < 0`, it gives good results. I just decided to keep the approach used in the book mainly because below `lon` is compared with `pi` so both checks must be consistent and I didn't analyse if the result should be ok. The naming is also taken from the book. In the same code there is a variable called `E` probably also taken from some source. In general I noticed that in Geometry names of variables are taken from papers even if they are against our guidelines. But sure, I already altered the algorithm so I can change the naming as well.

AFAIU using `math::equals()` or `math::smaller()` could add an error because the longitudes would be compared according to the EPS. So currently a value of `E` may be calculated and it may be not fully consistent with the value of `lon` and `pi`. But do you think this is a problem? Probably only if the value of `E` can changes rapidly and the later checks are there to reverse this change, but I think that this isn't the case. I'll think about it.